docs/workflow/testing/libraries/testing.md

```diff
@@ -1,54 +1,80 @@
 # Testing Libraries
 
-We use the OSS testing framework [xunit](https://github.com/xunit/xunit).
+## Full Build and Test Run
 
-To build the tests and run them you can call the libraries build script. For libraries tests to work, you must have built the coreclr or mono runtime for them to run on.
+These example commands automate the test run and all pre-requisite build steps in a single command from a clean enlistment.
 
-**Examples**
-- The following shows how to build only the tests but not run them:
+- Build all pre-requisites (clr + libs) and tests and run them using CoreCLR:
 ```
-build.cmd/sh -subset libs.tests
+build.cmd/sh -subset clr+libs+libs.tests -test
 ```
 
-- The following builds and runs all tests using clr:
+- Build all pre-requisites (mono + libs) and tests and run them using Mono:
 ```
-build.cmd/sh -subset clr+libs.tests -test
+build.cmd/sh -subset mono+libs+libs.tests -test
 ```
 
-- The following builds and runs all tests using mono:
+- Run all tests - Builds clr in release, libs+tests in debug:
 ```
-build.cmd/sh -subset mono+libs.tests -test
+build.cmd/sh -subset clr+libs+libs.tests -test -rc Release
 ```
 
-- The following builds and runs all tests in release configuration:
+- Run all tests - Build mono and libs for x86 architecture:
 ```
-build.cmd/sh -subset libs.tests -test -c Release
+build.cmd/sh -subset mono+libs+libs.tests -test -arch x86
 ```
 
-- The following builds clr in release, libs in debug and runs all tests:
+## Partial Build and Test Runs
+
+Doing full build and test runs takes a long time and is very inefficient if you need to iterate on a change.
+For greater control and efficiency individual parts of the build + testing workflow can be run in isolation.
+See the [Building instructions](../../building/libraries/README.md) for more info on build options.
+
+### Test Run Pre-requisites
+Before any tests can run we need a complete build to run them on. This requires building (1) a runtime, and
+(2) all the libraries. Examples:
+
+- Build release clr + debug libraries
 ```
-build.cmd/sh -subset clr+libs+libs.tests -test -rc Release
+build.cmd/sh -subset clr+libs -rc Release
 ```
 
-- The following builds mono and libs for x86 architecture and runs all tests:
+- Build release mono + debug libraries
 ```
-build.cmd/sh -subset mono+libs+libs.tests -test -arch x86
+build.cmd/sh -subset mono+libs -rc Release
 ```
 
-- The following example shows how to pass extra msbuild properties to ignore tests ignored in CI:
+Building the `libs` subset automatically copies all product binaries into a testhost folder in the bin directory.
+This is where the tests will load the binaries from during the run. If you later [rebuild any of product binaries](https://github.com/dotnet/runtime/blob/main/docs/workflow/building/libraries/README.md#iterating-on-systemprivatecorelib-changes)
+using the individual library build projects you must also build the `libs.pretest` subset to copy the new binaries to testhost.
```

Comment:
This isn't true. When building individual libraries, their output is copied into the testhost folder automatically. This infrastructure piece is called `binplacing`.

Reply:
That's why I expressed concerns about exposing the `libs.pretest` subset in this doc.

Reply:
Is the [existing documentation](https://github.com/dotnet/runtime/blob/8b116050be5735be8c94dcb8b0cd7554cab1e8dc/docs/workflow/building/libraries/README.md#iterating-on-systemprivatecorelib-changes) about incremental rebuild of S.P.C requiring libs.pretest inaccurate?

I can gladly update this but I am still attempting to distinguish between (a) it is a special case for S.P.C or (b) it isn't necessary in any circumstance.

Reply:
Right, CoreLib is special cased and doesn't binplace today. That should be fixed at some point but a concern was that its configuration need to be in sync with the native runtime bits.

cc @jkotas

Reply:
Hopefully the text now makes it clear that S.P.C is the special case and for everything else this is handled automatically.